### DISCUSSION TITLE
OSU why didn't publish on steam?

### DISCUSSION THREAD

**Discussion by Hime-Dev** [ppy/osu#18074] [Ideas] [answered]
Hello, there are players from all over the world, but why hasn't it been released on the steam market until now? Wouldn't it be better if it was published on the Steam market?

**Comment by peppy** [comment, marked as answer]
We get this request once in a while. Generally we won't push to steam unless there's a lot of demand.

**Comment by Schweeeeeeeeeeeeeeee** [comment]
We (Schweeeeeeeeeeeeeeee and Papertobi) suggest that osu!lazer should be made available on Steam. The game has now become stable and polished, and we believe it would greatly benefit from the accessibility that the Steam platform provides.

By adding the Steam Early Access tag, players would understand that the game is still in development. The ease of use provided by Steam would make the installation process simpler and more intuitive for new players.

Moreover, bringing osu!lazer to Steam would make it easier to play on devices such as the Steam Deck. The added support for controllers through Steam XInput would make the game more accessible to players who prefer using controllers over traditional mouse and keyboard controls.

In addition, the changelog for the game could also be made available on Steam, since people are familiar with the Steam changelog system. Additionally, there could be guides on the Steam community page to help new players get started with the game, such as a guide to move the mouse or set up the trackpad or learn streams.

By making osu!lazer available on Steam, we believe the game would be exposed to a much larger audience than it currently has, potentially expanding the player base and community.

This text was written and suggested by the osu!lazer Community members Schweeeeeeeeeeeeeeee and PaperTobi.

**Reply by peppy** [reply]
Thanks for your suggestions! I can't guarantee any movement on this issue as we've run this game stand-alone and independent until now, but do understand your concerns about accessibility, especially with the steam dick now being a thing.

Given that we are pushing towards mobile releases on app stores, they *may* be a future where it appears on steam. But the amount of work required to make this happen (and to maintain it) would be equal to a third mobile OS, so again, no guarantees.

**Reply by Hime-Dev** [reply]
I don't have any general information about publish a game to steam, all I know is a price of 100$ something. If there will be other expenses like this, osu will gladly help the players (at least I think so). I don't know if you need osu development for publish on steam market but osu lazer! publish on steam market wouldn't be bad. Thank you for everything.

**Reply by Schweeeeeeeeeeeeeeee** [reply]
https://github.com/ppy/osu/discussions/22694#discussioncomment-5377317
I meant that

It would be more accessible and controllers/updating (for linux :( ) would be supported out of the box as far as I know.

**Comment by Temanor** [comment]
I would also love to see a steam release now that lazer is starting to take over for stable.

**Comment by shps18ppt** [comment]
It wouldn't be worth it going on Steam as osu! has been a standalone software for almost 15+ years, and still gain popularity. And yes, although it would reach more players, it wouldn't be worth it as I believe peppy won't make use of the Steam API (achievements, multiplayer server, etc.). What peppy and the osu! team can do is make a separate build with the Steam APIs and can either use their own server or Steam's. I hope this comment can give a scenario of what would happen and why best not to put it on Steam. It does have a advantage and that is to port osu! to the Steam Deck, tho.

ps: uploading a game to steam is expensive as an indie game dev and tbh, wouldn't be worth it unless if its a whole gamedev studio. peppy can make it only compatible to the steam deck and force everyone to download it standalone if on pc/mac.

**Reply by ziv-vy** [reply]
It's $100 not $10000 so it's not that hard. The devs would take a minor hit on their next paycheck but nothing huge.

**Reply by Temanor** [reply]
> It's $100 not $10000 so it's not that hard. The devs would take a minor hit on their next paycheck but nothing huge.

100$ is very little when it comes to Osu. 
I think I alone have spent over 100$ on supporter and merch.
That being said, if Steam does takes a cut on supporter sales, I can understand if they want to avoid putting it there.

**Comment by rejedai** [comment]
+1 for steam. 

I believe that the game can even be sold, for a symbolic 1, 5, 10 dollars, additional profit will not be superfluous, while remaining free version on the site.

**Reply by ziv-vy** [reply]
Nope. Valve forces prices to be identical everywhere.

**Reply by Temanor** [reply]
> Nope. Valve forces prices to be identical everywhere.

Wasn't this only for steam keys?
"It is important that you don't give Steam customers a worse deal than Steam key purchasers"

I believe there might have been some emails saying otherwise in a recent lawsuit, but the user agreement doesn't say anything about this, so I would assume that was just a misunderstanding.

**Reply by ziv-vy** [reply]
Even then, making your game cost money on one platform and be free on another gives the player a bad impression. It's really scummy.
Imagine buying the game on Steam and realising that it was free on the site. I'd be **reallly** mad if that happened to me, especially considerring that I couldn't care less about what store front it is from.

**Reply by Temanor** [reply]
I agree. I think that if it is ever published on Steam, it should be free there as well.

**Reply by p1x3l101-10** [reply]
Im going to necropost here because I just want to clarify for future readers.

Judging by how Krita and Mindustry handle steam pricing, I think that it's perfectly ok to have the steam release cost money while having the game be free elsewhere.

Personally, I would like if the Osu steam release was free, but I don't mind paying $5 or whatever for the steam version. I like supporting devs of projects that I like.

**Reply by ziv-vy** [reply]
Charging people for something that they get for free elsewhere **_and_** them almost definetly not knowing that they could have gotten it elsewhere is scummy. The end.

Just because some other project did this does not mean that osu! should follow.

This feels like the devs saying "Steam players are richer. Make them pay more."

Also I think that you are drastically overestimating the amount of players who think that paying for free software just to have it on Steam is a good deal. You'd be shocked by how many osu! players don't even have steam.

**Comment by brunoaaveloso** [comment]
Would also love and buy Osu! day one on Steam. Hopefully devs reconsider a Steam launch.

**Comment by MinaPoppins** [comment]
It'd be great to see it on Steam

**Comment by ziv-vy** [comment]
I don't think that it is happening since the devs would have to pay a 30% tax of all Supporter purchases to Valve. 
Why would they pay that when they can just... not? People already go to the site to download osu! so it's unlikely to result in more players.

**Reply by Temanor** [reply]
Correct me if I'm wrong, but that is only for in-game purchases that utilizes the steam wallet.
Osu! supporter is only bought through the website anyways. There would be no need to pay any fee.

**Reply by ziv-vy** [reply]
This blatantly goes against the Steam TOS. You **have** to sell all microtransactions on Steam.

**Reply by LittleEndu** [reply]
You'll have to show actual quote from Steam ToS on that... because developers on Steam are saying they are allowed to sell their game and dlc and microtransactions on platforms other than Steam if the amount the consumer pays is the same on Steam and the other platform. osu! supporter is 4$/month so if that's the price on both osu.ppy.sh and s.team then publishing on Steam would be fine.

**Reply by ziv-vy** [reply]
...and why would anyone ever pay on a random website instead of steam when the price is identical? Pretty sure that they are not even allowed to mention the possibility of buying from outside of steam

edit: definetly yea. They are not even allowed to add links to osu.ppy.sh since there are microtransactions there.

**Reply by Qunas101** [reply]
I am pretty sure Steam allows selling anything you want however you want. The only restriction is that you cannot provide service that HAS to go through Steam for cheaper than on Steam. I.E. You can't generate a bunch of keys on Steam and sell it for cheaper somewhere else, because Steam then has to provide downloads and other services to users that gave them $0 income. You can sell your game for cheaper if you strip all Steam services out of it and provide downloads yourself

**Reply by ziv-vy** [reply]
I mean... the steamworks documentation seems pretty clear: [You **need** use the microtransaction API to allow users to make purchases via the steam wallet for **any** purchase.](https://partner.steamgames.com/doc/features/microtransactions)
<img width="1004" height="83" alt="image" src="https://github.com/user-attachments/assets/98844dbb-41fe-4f70-a9a5-bcf31735751c" />

And even if you still think that this does not mean that they can't avoid the cut, Valve isn't just gonna watch how their cut is bypassed. They'll just update their rules to explicitly ban this. There's no reason why valve would let a random indie game get the luxury of bypassing the cut with every other gaming publisher watching from the sidelines.

(note to maintainers of the repo: i somehow accidentally comented on the wrong place and it was offtopic there. I just copy pasted the comment where it was intended to be, please don't delete this comment for offtopic like the initial one)

**Comment by Qunas101** [comment]
Why though? The main reason FOSS gets published to Steam is for the convenience of automatic updates and maybe(?) to track your hours. osu! already does both of those things (unless you need tracking hours outside of gameplay for some reason)

**Reply by Temanor** [reply]
Steam overlay, Friends on Steam, Discoverability, etc.
There are plenty of reasons.

**Reply by MeerukYang** [reply]
> Steam overlay, Friends on Steam, Discoverability, etc. There are plenty of reasons.

you can add non-steam games into your steam library. in this way, steam overlay and friends can be used as normal. lazer works too.

![image](https://github.com/user-attachments/assets/9e5cc643-eaf6-46bc-bded-ac6b1c3ed361)
![image](https://github.com/user-attachments/assets/464cfb92-7d61-4d73-b3ff-91b3f8a88fe3)

you cannot invite your steam friends into a mutiplayer room though.

**Reply by ziv-vy** [reply]
Since a few updates ago, this is partially broken, and I have not reported it because they are not fixing it as this is not a supported scenario

Why? It creates another shortcut every time you try to hide the old osu! shortcut, and since windows will prefer the old one (in Windows search), you will never end up opening the steam version.

Also, Steam is stupid slow. Adding osu! to steam slows down boot time by a few seconds

**Reply by MeerukYang** [reply]
> Since a few updates ago, this is partially broken, and I have not reported it because they are not fixing it as this is not a supported scenario
> 
> Why? It creates another shortcut every time you try to hide the old osu! shortcut, and since windows will prefer the old one, you will never end up opening the steam version.
> 
> Also, Steam is stupid slow. Adding osu! to steam slows down boot time by a few seconds

instead of using a windows shortcut, you may use a steam shourtcut. that is via `steam://rungameid/xxxxx`. 

![image](https://github.com/user-attachments/assets/43e93a1a-c5e3-468e-9168-ea4845af63fd)

This indeed slows every boots but it can surely load steam overlay.

**Reply by ziv-vy** [reply]
There's also another issue: shift-tab toggles leaderboards.

**Reply by rejedai** [reply]
The main reason is that everything is in one place, even if you download 1 launcher through which the game is downloaded, it's much better than running around the internet and looking for where to buy/download, and come across a phishing site.

**Reply by Qunas101** [reply]
> running around the internet and looking for where to buy/download, and come across a phishing site.

Windows moment, I'm sorry. Hopefully they will publish to WinGet or Chocolatey


Also, Pro tip: if you add osu! as a non-steam game, you can add a custom background and logo from [steamgriddb](https://www.steamgriddb.com/game/4608) to make it less ugly (or make one yourself)

<details>
  <summary>I haven't updated mine since the "nipple" design, but you get the idea</summary>

![зображення](https://github.com/user-attachments/assets/b9fc7791-6b2f-4dcb-95d4-898c73e05ffc)

![зображення](https://github.com/user-attachments/assets/d1ddc900-ce5b-4e07-83a8-399bced9739b)

</details>
